src/Components/TableHeader/TableHeader.tsx

```diff
@@ -0,0 +1,61 @@
+import { Box, Button, Typography } from '@mui/material'
+import { Link, useLocation } from 'react-router-dom'
+import './TableHeader.module.scss'
+
+interface IProps {
+  title: string
+  subTitle: string
+  path:string
+}
+
+
+const TableHeader = ({ title, subTitle, path }: IProps) => {
+  const {pathname} = useLocation()
+  return <>
+    <Box component="header" className='header'>
+
+      <Box className='headerContainer'>
+
+        <Box className='leftSec'>
+          <Typography variant='subtitle2' className='biggerSize'>{title} Table Details</Typography>
+          <Typography variant='body2' className='smallerSize' >You can check all details</Typography>
+        </Box>
+
+        {pathname === "/dashboard/users"?"":
+        <Button color='primary' variant="contained" >
+        <Link to={path} className='linkStyle'>
```

Comment:
It's important to avoid nesting interactive elements, such as anchors (<a>), within buttons and vice versa. When working with HTML elements, it's crucial to choose the appropriate one based on its primary function. 

For instance, the use of <Link> within a <Button> may introduce nesting concerns. 
Typically, <a> elements are employed for navigation and linking, while buttons are reserved for performing actions. 

If a link needs a button-like appearance, it's recommended to achieve this through CSS styling without compromising its functionality. This ensures a clean and semantically correct structure in your code.